### PR TITLE
Use quelpa-autoremove-p when no prefix given

### PR DESCRIPTION
**Pull request by wyuenho**
Problem:

When `quelpa` is combined with `use-package-quelpa`, configuring the defcustom `quelpa-autoremove-p` will have no effect when the recipe does not explicitly supply the `:autoremove` keyword. This unintuitive and inconsistent behavior was introduced in #175, this PR fixes this issue.

### CODE REVIEW

File: quelpa.el [hunk 1]
```diff
@@ -1933,18 +1933,18 @@ to install.
...
+whether these actions should take place by default."
   (interactive (list nil))
   (run-hooks 'quelpa-before-hook)
   (when (quelpa-setup-p) ;if init fails we do nothing
     (let* ((arg (or arg
                     (let ((quelpa-melpa-recipe-stores
                            `(,@quelpa-melpa-recipe-stores ,quelpa-cache)))
                       (quelpa-interactive-candidate))))
-           (quelpa-upgrade-p (if current-prefix-arg t quelpa-upgrade-p)) ;shadow `quelpa-upgrade-p'
-           (quelpa-stable-p quelpa-stable-p) ;shadow `quelpa-stable-p'
```
Comment (kiennq) [inline quelpa.el:1946]: You shouldn't remove the `quelpa-stable-p` check here. It's for shadowing `quelpa-stable-p` so nested call will not modify the global `quelpa-stable-p`

**Comment by kiennq** [comment]
@wyuenho You can see the details why the current behavior is like that in this thread https://github.com/quelpa/quelpa/issues/169.
The initial implementation is to make sure that once user turn off the auto-remove, Quelpa will fallback to its previous behavior of not removing old package (for whatever reason).
Now after sometimes the feature is default and we don't see user reporting, I guess we can make the auto-remove become actually forced for upgrading case and only `quelpa` with auto-remove off can install package without deleting previous version.

**Comment by kiennq** [comment]
Also the test's failing, can you take a look?

File: quelpa.el (hunk 1 above)
Comment (wyuenho) [reply quelpa.el:1935]: Thanks. I looked at the code again. I must say, this is some of the most egregious abuse of dynamic scoping I've ever seen :P

**Review by kiennq — APPROVED** [review]

**Pull request merged**